src/Common/src/TypeSystem/Common/FieldDesc.FieldLayout.cs

```diff
@@ -43,7 +43,7 @@ public virtual bool HasGCStaticBase
             get
             {
                 if (!FieldType.IsValueType)
-                    return true;
+                    return !FieldType.IsPointer;
```

Comment:
This does not look right. It works, but it is putting a lot more stuff into the GC statics region than needed. I think that the implementation of this method should be:

```
var fieldType = this.FieldType;

if (fieldType.IsValueType)
    return ((DefType)fieldType).ContainsPointers;
else
    return fieldType.IsObjRef;
```